### PR TITLE
Update API specs

### PR DESCRIPTION
**Pull request by dnaka91**
Completely updating the API specs.
I also replaced the JS/TS interfaces with tables to make it language agnostic.

### CODE REVIEW

File: api.spec.md [hunk 1]
```diff
@@ -1,70 +1,141 @@
...
 
-## Event Types
-Generic Event, sent as json to the hub
-```ts
-interface EventType<T> {
-  event: string,
-  data: T; // one of the following
-}
-```
+- [API v0.1.4](#api-v014)
```
Comment (negue) [inline api.spec.md]: I think this line can be removed. no real need to jump 2 lines higher

I thought there was a github-flavoured markdown feature to create the table of content but I couldn't find it.

**Review by negue — CHANGES_REQUESTED** [review]
only one issue, but other than that, nice :>

File: api.spec.md (hunk 1 above)
Comment (dnaka91) [reply api.spec.md]: Yeah I know, it's kinda useless to put it but the table of contents feature of the V Code Plugin auto-generates and then updates the table on each save. I will also look for the GitHub Markdown feature.

Let me see if there is an option for it.

Do we even need a toc I wonder? The spec is not that long yet.

Comment (negue) [reply api.spec.md]: True, maybe the toc is a bit too much for now anyway^^, lets remove it for now

Comment (dnaka91) [reply api.spec.md]: And done. Also found some smaller spelling mistakes and fixed them.

**Pull request merged**